test/run-test.py

```diff
@@ -25,39 +26,50 @@ def compare_structures(data1, data2):
         return (data1 == data2)
     return True
 
+def perform_request(test_data, url):
+    if test_data['method'] == 'GET':
+        r = requests.get(url)
+    elif test_data['method'] == 'PUT':
+        r = requests.put(url)
+    elif test_data['method'] == 'POST':
+        r = requests.post(url)
+    elif test_data['method'] == 'DELETE':
+        r = requests.delete(url)
+    else:
+        print('Unsupported HTTP method: %s' % test_data['method'])
+        return False
+
+    if r.status_code != test_data['responseCode']:
+        print('Got response code %d, expected %d' % (r.status_code, test_data['responseCode']))
+        return False
+
+    if 'responseText' in test_data:
+        if r.text != test_data['responseText']:
+            print('Got response text:\n\n%s\n\nexpected:\n\n%s' % (r.text, test_data['responseText']))
+            return False
+    elif 'responseJsonFile' in test_data:
+        response_data = json.load(open('%s/%s' % (TEST_DIR, test_data['responseJsonFile'])))
+        if not compare_structures(r.json(), response_data):
+            print('Got response JSON:\n\n%s\n\nexpected:\n\n%s' % (r.text, json.dumps(response_data)))
+            return False
+    return True
+
 ENDPOINT = sys.argv[1]
 TEST_DIR = sys.argv[2]
 
 test_data = json.load(open('%s/metadata.json' % TEST_DIR))
 
+retries = test_data.get('retries', 1)
```

Comment:
I would make this default to `0` and modify the logic below.

Reply:
changed